File: tests/python/arv-camera-test.py
#!/usr/bin/env python

#  If you have installed aravis in a non standard location, you may need
#   to make GI_TYPELIB_PATH point to the correct location. For example:
#
#   export GI_TYPELIB_PATH=$GI_TYPELIB_PATH:/opt/bin/lib/girepositry-1.0/
#
#  You may also have to give the path to libaravis.so, using LD_PRELOAD or
#  LD_LIBRARY_PATH.

import sys
import gi

gi.require_version ('Aravis', '0.8')

from gi.repository import Aravis

Aravis.enable_interface ("Fake")

try:
    if len(sys.argv) > 1:
        camera = Aravis.Camera.new (sys.argv[1])
    else:
        camera = Aravis.Camera.new (None)
except TypeError:
	print ("No camera found")
	exit ()

camera.set_region (0,0,128,128)
camera.set_frame_rate (10.0)
camera.set_pixel_format (Aravis.PIXEL_FORMAT_MONO_8)

payload = camera.get_payload ()

[x,y,width,height] = camera.get_region ()

print ("Camera vendor : %s" %(camera.get_vendor_name ()))
print ("Camera model  : %s" %(camera.get_model_name ()))
print ("ROI           : %dx%d at %d,%d" %(width, height, x, y))
print ("Payload       : %d" %(payload))
print ("Pixel format  : %s" %(camera.get_pixel_format_as_string ()))

stream = camera.create_stream (None, None)

for i in range(0,10):
	stream.push_buffer (Aravis.Buffer.new_allocate (payload))

print ("Start acquisition")

camera.start_acquisition ()

print ("Acquisition")

for i in range(0,20):
	image = stream.pop_buffer ()
	print (image)
	if image:
		stream.push_buffer (image)

print ("Stop acquisition")

camera.stop_acquisition ()

File: tests/js/arv-camera-test.js
#!/usr/bin/env gjs

/* SPDX-License-Identifier:Unlicense */

/*
   If you have installed aravis in a non standard location, you may need
   to make GI_TYPELIB_PATH point to the correct location. For example:

   export GI_TYPELIB_PATH=$GI_TYPELIB_PATH:/opt/bin/lib/girepositry-1.0/

   You may also have to give the path to libaravis.so, using LD_PRELOAD or
   LD_LIBRARY_PATH.
 */

const GLib = imports.gi.GLib;
const Aravis = imports.gi.Aravis;

let camera = Aravis.Camera.new (null);

camera.set_region (0,0,128,128);
camera.set_pixel_format (Aravis.PIXEL_FORMAT_MONO_8);
camera.set_frame_rate (10.0);

let [x,y,width,height] = camera.get_region ();
let payload = camera.get_payload ();

print ("Camera vendor : ", camera.get_vendor_name ());
print ("Camera model  : ", camera.get_model_name ());
print ("ROI           : ", width, "x", height, " at ", x, ",", y);
print ("Payload       : ", payload);
print ("Pixel format  : ", camera.get_pixel_format_as_string ());

let stream = camera.create_stream (null);

for (let i = 0; i < 10; i++) {
    stream.push_buffer (Aravis.Buffer.new_allocate (payload));
}

print ("Start Acquisition");

camera.start_acquisition ();

print ("Acquisition");

for (let i = 0; i < 20; i++) {
    let buffer = stream.pop_buffer ();
    print (buffer);
    if (buffer) {
	stream.push_buffer (buffer);
    }
}

print ("Stop acquisition");

camera.stop_acquisition ();
